client/v2/internal/tx/context.go

```diff
@@ -0,0 +1,458 @@
+package tx
+
+import (
+	"bufio"
+	"context"
+	"cosmossdk.io/client/v2/tx"
+	"cosmossdk.io/core/address"
+	"encoding/json"
+	"fmt"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/gogoproto/proto"
+	"github.com/spf13/viper"
+	"google.golang.org/grpc"
+	"io"
+	"os"
+	"path"
+	"sigs.k8s.io/yaml"
+	"strings"
+)
+
+// PreprocessTxFn defines a hook by which chains can preprocess transactions before broadcasting
+type PreprocessTxFn func(chainID string, key keyring.KeyType, tx tx.TxBuilder) error
+
+// Context implements a typical context created in SDK modules for transaction
+// handling and queries.
+type Context struct {
```

Comment:
Do we really need all those information? Like From, FromName and FromAddress could only be set to one thing (and when using a from name we directly query the keyring -- if possible).
This looks like mostly the exact same one that we have currently